.github/workflows/checklinks.yaml

```diff
@@ -0,0 +1,37 @@
+name: CheckLinks
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  checklinks:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v3
+      - name: Ruby
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: 3.1
+      - name: Setup Rubygems, Bundler, jekyll
+        run: |
+          gem update --system --no-document
```

Comment:
This might be overkill. Whatever version of rubygems that shipped with the installed ruby is probably fine.